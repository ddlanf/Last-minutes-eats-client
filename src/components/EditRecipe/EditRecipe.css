.edit-recipe{
    width: 100%;
    max-width: 1000px;
    margin-left: auto;
    margin-right: auto;
    margin-top: 8%;
    text-align: center;
}

.edit-recipe-heading{
    font-size: 5vw;
    margin-top: 2%;
}

.edit-recipe-form{
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 10%;
}

.edit-recipe-label{
    display: block;
    font-size: 4.5vw;
    margin-top: 1vw;
    margin-bottom: 2%;
}

.edit-recipe-input{
    width: 80%;
    margin-bottom: 5%;
    font-size: 4.5vw;
    max-width: 500px;
}

.edit-recipe-ingredients-and-steps{
    width: 100%;
    font-size: 3vw;
}

.edit-recipe-ingredients-and-steps > *{
    font-size: 3vw;
}

.edit-recipe-button-and-text{
    width: 80%;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

.edit-recipe-variable-input{
    width: 80.3%;
    font-size: 4vw;
}

.edit-recipe-plus{
    position: absolute;
    width: 10%;
    height: 100%;
    left: 0;
}

.edit-recipe-minus{
    position: absolute;
    width: 10%;
    height: 100%;
    right: 0;
}

.edit-recipe-submit{
    margin-left: auto;
    margin-right: auto;
    margin-top: 5%;
    margin-bottom: 5%;
    width: 25%;
}

@media screen and (min-width: 1000px) {

    .edit-recipe-ingredients-and-steps{
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: flex-start;
    }

    .edit-recipe-label{
        display: block;
        font-size: 3.5vw;
        margin-top: 1vw;
    }

    .edit-recipe-input{
        font-size: 3vw;
    }

    .edit-recipe-submit{
        height: 3vw;
    }

    .edit-recipe-variable-input{
        font-size: 3vw;
    }

}