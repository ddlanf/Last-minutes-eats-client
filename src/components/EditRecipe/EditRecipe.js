import React, { Component } from 'react'
import './EditRecipe.css'

export default class EditRecipe extends Component {
    render() {
        return (
            <div>
                
            </div>
        )
    }
}