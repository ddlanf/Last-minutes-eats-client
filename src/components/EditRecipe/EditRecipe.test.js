import React from 'react';
import ReactDOM from 'react-dom';
import EditRecipe from './EditRecipe';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from '../../reducers/index'

describe('EditRecipe component', () => {

   const store = createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>   
                      <Provider store={store}>
                        <EditRecipe/>
                      </Provider>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
          <BrowserRouter>   
            <Provider store={store}>
              <EditRecipe/>
            </Provider>
          </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});