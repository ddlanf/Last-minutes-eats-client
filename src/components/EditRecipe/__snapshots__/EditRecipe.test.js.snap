// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecipe component renders the UI as expected 1`] = `
Array [
  <section
    className="edit-recipe-gate"
  >
    <form
      className="edit-recipe-gate-form "
      onSubmit={[Function]}
    >
      <label
        className="edit-recipe-gate-label"
      >
        Enter token to edit this recipe
      </label>
      <input
        className="edit-recipe-gate-input"
        name="token"
        onChange={[Function]}
        type="password"
      />
      
      <button
        className="edit-recipe-gate-submit"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
      <a
        className="edit-recipe-gate-back"
        href="/view-recipe/undefined"
        onClick={[Function]}
      >
        Back
      </a>
    </form>
  </section>,
  <section
    className="edit-recipe"
  >
    <h1
      className="edit-recipe-heading"
    >
      Edit This Recipe
    </h1>
    <form
      className="edit-recipe-form"
      onSubmit={[Function]}
    >
      <div
        className="edit-recipe-top-field"
      >
        <div
          className="edit-recipe-name-field"
        >
          <label
            className="edit-recipe-label"
          >
            Recipe Name
          </label>
          <input
            className="edit-recipe-input"
            defaultValue=""
            name="recipe_name"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="edit-recipe-image-field"
        >
          <label
            className="edit-recipe-label"
          >
            Recipe Image Link
          </label>
          <input
            className="edit-recipe-input"
            defaultValue=""
            name="image"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
      <div
        className="edit-recipe-ingredients-and-instructions"
      >
        <div
          className="edit-recipe-instructions-field"
        >
          <label
            className="edit-recipe-label"
          >
            Instructions
          </label>
          <div
            className="edit-recipe-instructions-input-box"
          >
            <div
              className="edit-recipe-button-and-text steps"
              style={
                Object {
                  "display": "block",
                }
              }
            >
              <span
                className="edit-recipe-number"
              >
                1
              </span>
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <textarea
                className="edit-recipe-variable-input"
                defaultValue=""
                name="step-1"
                onChange={[Function]}
                type="text"
              />
              
            </div>
            <div
              className="edit-recipe-button-and-text steps"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <span
                className="edit-recipe-number"
              >
                2
              </span>
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <textarea
                className="edit-recipe-variable-input"
                defaultValue=""
                name="step-2"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="edit-recipe-button-and-text steps"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <span
                className="edit-recipe-number"
              >
                3
              </span>
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <textarea
                className="edit-recipe-variable-input"
                defaultValue=""
                name="step-3"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="edit-recipe-button-and-text steps"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <span
                className="edit-recipe-number"
              >
                4
              </span>
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <textarea
                className="edit-recipe-variable-input"
                defaultValue=""
                name="step-4"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="edit-recipe-button-and-text steps"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <span
                className="edit-recipe-number"
              >
                5
              </span>
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <textarea
                className="edit-recipe-variable-input"
                defaultValue=""
                name="step-5"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="edit-recipe-button-and-text steps"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <span
                className="edit-recipe-number"
              >
                6
              </span>
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <textarea
                className="edit-recipe-variable-input"
                defaultValue=""
                name="step-6"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="edit-recipe-button-and-text steps"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <span
                className="edit-recipe-number"
              >
                7
              </span>
              
              <textarea
                className="edit-recipe-variable-input"
                defaultValue=""
                name="step-7"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
        <div
          className="edit-recipe-ingredients-field"
        >
          <label
            className="edit-recipe-label"
          >
            Ingredients
          </label>
          <div
            className="edit-recipe-ingredients-input-box"
          >
            <div
              className="edit-recipe-button-and-text"
              style={
                Object {
                  "display": "block",
                }
              }
            >
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <input
                className="edit-recipe-variable-input"
                defaultValue=""
                name="ingredient-1"
                onChange={[Function]}
                type="text"
              />
              
            </div>
            <div
              className="edit-recipe-button-and-text"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <input
                className="edit-recipe-variable-input"
                defaultValue=""
                name="ingredient-1"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="edit-recipe-button-and-text"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <input
                className="edit-recipe-variable-input"
                defaultValue=""
                name="ingredient-1"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="edit-recipe-button-and-text"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <button
                className="edit-recipe-plus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                +
              </button>
              <input
                className="edit-recipe-variable-input"
                defaultValue=""
                name="ingredient-1"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="edit-recipe-button-and-text"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              
              <input
                className="edit-recipe-variable-input"
                defaultValue=""
                name="ingredient-1"
                onChange={[Function]}
                type="text"
              />
              <button
                className="edit-recipe-minus"
                onClick={[Function]}
                style={
                  Object {
                    "display": "block",
                  }
                }
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
      </div>
      <label
        className="edit-recipe-preparation-time-label"
      >
        Preparation Time
      </label>
      <div
        className="edit-recipe-preparation-time-box"
      >
        <input
          className="edit-recipe-preparation-time-input"
          defaultValue=""
          name="preparation_time"
          onChange={[Function]}
          type="number"
        />
        <select
          className="edit-recipe-preparation-time-unit-input"
          onChange={[Function]}
          value="minutes"
        >
          <option
            value="seconds"
          >
            sec
          </option>
          <option
            value="minutes"
          >
            min
          </option>
        </select>
      </div>
      
      <button
        className="edit-recipe-submit"
        disabled={true}
        name="submit"
        style={
          Object {
            "opacity": "0.5",
          }
        }
        type="submit"
      >
        Submit
      </button>
      <button
        className="edit-recipe-back"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </form>
  </section>,
]
`;
